### ISSUE TITLE
Debug flag to `run_language_modeling` triggers error

### ISSUE THREAD

**Issue by dmlap** [huggingface/transformers#6308]
## Environment info

- `transformers` version: 3.0.2
- Platform: Linux-5.4.0-42-generic-x86_64-with-glibc2.29
- Python version: 3.8.2
- PyTorch version (GPU?): 1.6.0+cu101 (True)
- Tensorflow version (GPU?): not installed (NA)
- Using GPU in script?: yes, run_language_modeling.py
- Using distributed or parallel set-up in script?: no

### Who can help
 
I'd guess @sgugger or @julien-c

## Information
I'm using [run_language_modeling.py](https://github.com/huggingface/transformers/blob/master/examples/language-modeling/run_language_modeling.py) and turned on debug output to double check things were working as I expected. Unfortunately, [trainer.py](https://github.com/huggingface/transformers/blob/master/src/transformers/trainer.py#L628) keys off that debug option to invoke `xm.master_print(...)` and `xm`/`torch_xla.core.xla_model` isn't loaded because I'm not working on a TPU-based system.

The problem arises when using:
* [X] the official example scripts: (give details below)
* [ ] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [X] my own task or dataset: (give details below)

## To reproduce

All steps should be run on a system with a GPU but no TPU. Steps to reproduce the behavior:

1. Run `run_language_modeling.py` with the debug flag:
```sh
python run_language_modeling.py \
  --output_dir ./output \
  --model_type gpt2 \
  --model_name_or_path gpt2 \
  --do_train \
  --train_data_file ./train.txt \
  --learning_rate 1e-4 \
  --num_train_epochs 1 \
  --save_total_limit 2 \
  --save_steps 200 \
  --do_eval \
  --eval_data_file ./eval.txt \
  --debug
```
2. Allow the script to run.

The command will error towards the end with this traceback:
```sh
Epoch:   0%|                                                                                                                                                   | 0/1 [54:06<?, ?it/s]
Traceback (most recent call last):
  File "run_language_modeling.py", line 281, in <module>
    main()
  File "run_language_modeling.py", line 245, in main
    trainer.train(model_path=model_path)
  File "/home/user/project/env/lib/python3.8/site-packages/transformers/trainer.py", line 570, in train
    xm.master_print(met.metrics_report())
NameError: name 'xm' is not defined
```
## Expected behavior

The script exits without error.

**Comment by sgugger** [comment]
The documentation of `debug` clearly points out this parameter is only used for TPU-training:
```
debug (:obj:`bool`, `optional`, defaults to :obj:`False`):
            When training on TPU, whether to print debug metrics or not.
```
It's called simply debug (and not `tpu_debug`) for harmonization with `TFTrainer`.

**Comment by dmlap** [comment]
Ah, my bad for not fully reading the documentation. Would you be open to a PR with a guard or more-specific error message for this scenario?

**Comment by sgugger** [comment]
Sure!

**Comment by LysandreJik** [comment]
closed by #6390
